ompy/error_finder.py

```diff
@@ -0,0 +1,433 @@
+                be used rather than the mean. Default is False.
+            full (bool, optional): If the sample trace should
+                be returned or not.
+        Returns:
+            The relative error of the nuclear level density and
+            gamma strength function.
+            Optionally returns the full sample trace if `trace` is true.
+
+        Raises:
+            ValueError: If length of `nlds` doesn't match the length of `gsfs`.
+            pyMC3 may raise errors. Please report if such errors occurs.
+        """
+        algo = None
+        if self.algorithm == 'lin':
+            algo = self.linear
+        elif self.algorithm == 'log':
+            algo = self.logarithm
+        else:
+            raise NotImplementedError(
+                f"Algorithm '{algorithm}' not implemented")
+
+        return algo(*self.condition_data(nlds, gsfs),
+                    median=median, full=full)
+
+    def logarithm(self, E_nld: ndarray, E_gsf: ndarray,
+                  nld_obs: ndarray, gsf_obs: ndarray,
+                  coef_nld: ndarray, coef_gsf: ndarray,
+                  vals_nld: ndarray, vals_gsf: ndarray,
+                  median: Optional[bool] = False, full: Optional[bool] = False
+                  ) -> Union[Tuple[Vector, Vector], Any]:
+        """ Calculate the relative errors of the NLD and γSF points using a
+        logarithmic model.
+
+        Args:
+            E_nld (ndarray): Energy points of the NLD
+            E_gsf (ndarray): Energy points of the γSF
+            nld_obs (ndarray): The observation tensor of the NLD
+            gsf_obs (ndarray): The observation tensor of the γSF
+            coef_nld (ndarray): Indices needed for broadcasting
+            coef_gsf (ndarray): Indices needed for broadcasting
+            vals_nld (ndarray): Indices needed for broadcasting
+            vals_gsf (ndarray): Indices needed for broadcasting
+            median (bool, optional): If the resulting relative errors should be
+                the mean or the median.
+            full (bool, optional): If the trace from the pyMC3 sampling should
+                be returned. Useful for debugging.
+        """
+
+        N, _, M_nld = nld_obs.shape
+        _, _, M_gsf = gsf_obs.shape
+
+        self.LOG.info("Starting pyMC3 inference - logarithmic model")
+
+        with pm.Model() as model:
+
+            σ_D = pm.HalfFlat("σ_D", testval=self.testvals['σ_D'])
+            σ_F = pm.HalfFlat("σ_F", testval=self.testvals['σ_F'])
+            σ_α = pm.HalfFlat("σ_α", testval=self.testvals['σ_α'])
+
+            D = pm.Normal("D", mu=0., sigma=σ_D, shape=[N, N-1])[:, coef_nld]
+            F = pm.Normal("F", mu=0., sigma=σ_F, shape=[N, N-1])[:, coef_gsf]
+            α = pm.Normal("α", mu=0., sigma=σ_α, shape=[N, N-1])
+
+            σ_ρ = FermiDirac("σ_ρ", lam=10., mu=1., shape=M_nld)[vals_nld]
+            σ_f = FermiDirac("σ_f", lam=10., mu=1., shape=M_gsf)[vals_gsf]
```

Comment:
/If/ the parametrization turns out t have something to say, you might choose to let `lam` and `mu` be instance attributes